src/test/javascript/spec/component/exam/exam-scores/exam-scores-average-scores-graph.component.spec.ts

```diff
@@ -105,4 +114,44 @@ describe('ExamScoresAverageScoresGraphComponent', () => {
         expect(component.ngxData).toEqual(expectedData);
         expect(component.ngxColor.domain).toEqual(expectedColorDomain);
     };
+
+    describe('test exercise navigation', () => {
+        const event = { name: 'test', value: 3 };
+        it('should navigate if event is valid', () => {
+            component.lookup['test'] = { exerciseId: 42, exerciseType: ExerciseType.QUIZ };
+
+            component.onSelect(event);
+
+            expect(navigateToExerciseMock).toHaveBeenCalledOnce();
+            expect(navigateToExerciseMock).toHaveBeenCalledWith(42, ExerciseType.QUIZ);
+        });
+
+        it('should not navigate if exercise id is missing', () => {
+            component.lookup['test'] = { exerciseType: ExerciseType.QUIZ };
+
+            component.onSelect(event);
+
+            expect(navigateToExerciseMock).toHaveBeenCalledTimes(0);
+        });
+
+        it('should navigate if exercise type is missing', () => {
```

Comment:
Shouldn't this be 'should **not** navigate if exercise type is missing' ?

Reply:
Good catch. Fixed. :)